src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -1984,6 +1985,22 @@ public Mono<Void> moveModifications(UUID groupUuid, UUID before, List<UUID> modi
         return Mono.fromRunnable(() -> networkModificationRepository.moveModifications(groupUuid, modificationsToMove, before));
     }
 
+    public Mono<Void> duplicateModifications(UUID targetGroupUuid, UUID sourceGroupUuid, List<UUID> modificationsToDuplicate) {
+        // This function cannot be @Transactional because we clone all modifications resetting their id to null,
+        // which is not allowed by JPA if we still stay in the same Tx.
+        List<ModificationEntity> newModificationList = new ArrayList<>();
+        for(UUID modifyId : modificationsToDuplicate) {
+            Optional<ModificationEntity> clone = this.modificationRepository.findById(modifyId);
```

Comment:
I decided to keep it this way because I want to detect all bad modification uuid (they must be returned).
Selecting a modification from id is very fast, so it should not be a performance issue to have this for loop.

Reply:
for (UUID modifyId : modificationsToDuplicate) {
            modificationRepository.findById(modifyId).ifPresentOrElse(newModificationList::add, () -> modificationsInFailure.add(modifyId));
        }

Reply:
In your example, clone.setid(null) is missing.
With a bloc of code, it becomes not easier to write and read.
```
for (UUID modifyId : modificationsToDuplicate) {
            modificationRepository.findById(modifyId).ifPresentOrElse(clone -> {
                clone.setId(null); newModificationList.add(clone);
            }, () -> missingModificationList.add(modifyId));
        }
```